pkg/virt-handler/isolation/detector.go

```diff
@@ -159,6 +159,47 @@ func (s *socketBasedIsolationDetector) AdjustResources(vm *v1.VirtualMachineInst
 	return nil
 }
 
+const qemuProcessExecutable = "qemu-kvm"
+
+// AdjustQemuProcessMemoryLimits adjusts QEMU process MEMLOCK rlimits that runs inside
+// virt-launcher pod on the given VMI according to its spec.
+// Only VMI's with VFIO devices (e.g: SRIOV, GPU) require QEMU process MEMLOCK adjustment.
+func AdjustQemuProcessMemoryLimits(podIsoDetector PodIsolationDetector, vmi *v1.VirtualMachineInstance) error {
+	if !util.IsVFIOVMI(vmi) {
+		return nil
+	}
+
+	isolationResult, err := podIsoDetector.Detect(vmi)
+	if err != nil {
+		return err
+	}
+
+	processes, err := ps.Processes()
+	if err != nil {
+		return fmt.Errorf("failed to get all processes: %v", err)
+	}
+	qemuProcess := findIsolatedQemuProcess(processes, isolationResult.PPid())
```

Comment:
What if [no-fork](https://github.com/kubevirt/kubevirt/blob/b2f3a6520ab821dbdf7a7a15f3da24bc32ed449a/cmd/virt-launcher/virt-launcher.go#L345) is `true`. Would the virt-launcher process and the qemu one still have the same ppid?

Reply:
That flag is a hack, we can reach here only in one condition. The binary is called without the  flag and then it calls itself with the flag (so not recursion will occur).

Reply:
No, when no-fork is `true` there will be no extra virt-launcher process unlike when no-fork is `false`,
virt-launcher and qmeu processes will not have the same PPID.

There was a dissolution about this flag with more details about virt-launcher processes state
https://github.com/kubevirt/kubevirt/pull/5558#discussion_r629351421